test/fx/test_fx_param_shape_control_flow.py

```diff
@@ -0,0 +1,141 @@
+import unittest
+import torch
+import torch.fx
+
+
+class MyModuleBase(torch.nn.Module):
+    def forward(self, x):
+        matrx = self.get_mul_matrix()
+        if self.no_relu():
+            return torch.mm(x, matrx)
+        else:
+            return torch.relu(torch.mm(x, matrx))
+
+    def get_mul_matrix(self):
+        return self.param
+
+    def no_relu(self):
+        raise Exception("not implemented")
+
+class MyModuleParamShape(MyModuleBase):
+    def __init__(self, in_channels):
+        super().__init__()
+        self.param = torch.nn.Parameter(torch.randn(in_channels, 3))
+
+    def no_relu(self):
+        return self.param.shape[0] < 10
+
+
+class MyModuleParamSize(MyModuleBase):
+    def __init__(self, in_channels):
+        super().__init__()
+        self.param = torch.nn.Parameter(torch.randn(in_channels, 3))
+
+    def no_relu(self):
+        return self.param.size()[0] < 10
+
+
+class MyModuleParamDim(MyModuleBase):
+    def __init__(self, param):
+        super().__init__()
+        self.param = param
+
+    def get_mul_matrix(self):
+        return self.param[0] if (self.param.dim() == 3) else self.param
+
+    def no_relu(self):
+        return self.param.dim() == 3
+
+
+class MyModuleParamNDim(MyModuleBase):
+    def __init__(self, param):
+        super().__init__()
+        self.param = param
+
+    def get_mul_matrix(self):
+        return self.param[0] if (self.param.ndim == 3) else self.param
+
+    def no_relu(self):
+        return self.param.ndim == 3
+
+
+class MyModuleParamNumEl(MyModuleBase):
+    def __init__(self, in_channels):
+        super().__init__()
+        self.param = torch.nn.Parameter(torch.randn(in_channels, 3))
+
+    def no_relu(self):
+        return self.param.numel() < 10 * 3
+
+
+
+class MyModuleParamNElement(MyModuleBase):
+    def __init__(self, in_channels):
+        super().__init__()
+        self.param = torch.nn.Parameter(torch.randn(in_channels, 3))
+
+    def no_relu(self):
+        return self.param.nelement() < 10 * 3
+
+
+
+class TestConstParamShapeInControlFlow(unittest.TestCase):
+
+    def verify_mm_relu_mods(self, mm_only_mod, relu_mod):
+        """
+        Verify one module only does a mm op while the other
+        performs both mm and relu ops in cascade
+        """
+        x = torch.randn(10, 5)
+        torch.testing.assert_allclose(mm_only_mod(x), torch.mm(x, mm_only_mod.get_mul_matrix()))
```

Comment:
We should do this value check on a `GraphModule` constructed from each `traced_graph*` as well to make sure that the generated code matches with the original Python code